lib/pbio/src/light/animation.c

```diff
@@ -120,6 +120,7 @@ PROCESS_THREAD(pbio_light_animation_process, ev, data) {
         clock_time_t interval = animation->next(animation);
         if (pbio_light_animation_is_started(animation)) {
             etimer_reset_with_new_interval(&animation->timer, interval);
+            etimer_restart(&animation->timer);
```

Comment:
We are already calling `etimer_reset_with_new_interval()` here, why do we also need `etimer_restart()`? It seems redundant.

Reply:
I've never done any Contiki programming before so this might be wrong, but according to their docs, `etimer_reset()` sets the timer to expire from the previous expire time whereas `etimer_restart()` restarts the timer from the current time. Looking at the implementation for `etimer_reset_with_new_interval()`, it internally calls `etimer_reset()` (as you'd expect). I suspect with a low interval, it is possible that the new timer is set to fire in the past, which seems to cause problems.